### ISSUE TITLE
When using blueprint's create, association records are not being populated

### ISSUE THREAD

**Issue by leobreakneck** [balderdashy/sails#3375]
When you use POST /:model, it should return the populated values from associations, according to the docs: http://sailsjs.org/documentation/reference/blueprint-api/create
However it only returns the id of the association. When using GET, PUT and DELETE the values are being populated. I'm using sails v0.11.2 and sails.blueprints.populate is set to true. This can be easily fixed by overriding the create action, but I believe it should be the default behavior.

**Comment by crobinson42** [comment]
You can turn populate on or off in blueprints config options.

**Comment by leobreakneck** [comment]
Populate is turned on. But it doesn't works when you create a new record, only for the other methods.

**Comment by jdcauley** [comment]
@leobreakneck can you provide any more information? Such as examples of your requests?

**Comment by leobreakneck** [comment]
An example using default blueprints, and populate set to true.
 
There is a User and Pet model:
User:
```
module.exports = {
  attributes: {
	name:'STRING',
        age:'INTEGER',
        pets:{
            collection: 'pet',
            via: 'owner'
        }
  }
};
```
Pet:
```
module.exports = {

  attributes: {
	name:'STRING',
        color:'STRING',
        owner:{
            model:'user'
        }
  }
};
```
POST /user?name=Jon&age=20 returns:
```
{
  "name": "Jon",
  "age": 20,
  "createdAt": "2015-11-25T22:55:45.789Z",
  "updatedAt": "2015-11-25T22:55:45.789Z",
  "id": 2
}
```
POST /pet?name=Bingo&color=white&owner=2 returns
```
{
  "name": "Bingo",
  "color": "white",
  "owner": 2,
  "createdAt": "2015-11-25T22:59:22.420Z",
  "updatedAt": "2015-11-25T22:59:22.420Z",
  "id": 5
}
```
The owner is not populated. On the other hand, when you use
PUT /pet/5?color=brown it returns
```
{
  "owner": {
    "name": "Jon",
    "age": 20,
    "createdAt": "2015-11-25T22:55:45.789Z",
    "updatedAt": "2015-11-25T22:55:45.789Z",
    "id": 2
  },
  "name": "Bingo",
  "color": "brown",
  "createdAt": "2015-11-25T22:59:22.420Z",
  "updatedAt": "2015-11-25T23:01:09.832Z",
  "id": 5
}
```
The owner is populated. 
Please let me know if you need more information.

**Comment by RWOverdijk** [comment]
@leobreakneck It's not "wrong" that it doesn't populate, but I do agree that some normalization is desired. This can be fixed by changing the default blueprint (and simply adding populate), or by overriding it in your own project.

**Comment by leobreakneck** [comment]
@RWOverdijk , I do agree with you. I fixed this in my project by extending blueprint's create, however I opened this issue because I believed it should be the normal behavior, since it happens on every other action.

**Comment by RWOverdijk** [comment]
@leobreakneck Any other day I'd agree with you. However, this is a breaking change so we'd have to up a major version. We'd have to ask management about that :p

**Comment by sailsbot** [comment]
Thanks for posting, @leobreakneck.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by remzr7** [comment]
This issue seems unresolved

**Comment by leobreakneck** [comment]
Hi @remzr7 ,
I got this working by doing the following:
Add a file named create.js to api/blueprints. This code will replace the original blueprints create action.

`/**
 * Module dependencies
 */
var actionUtil = require('sails/lib/hooks/blueprints/actionUtil');

/**
 * Create Record
 *
 * post /:modelIdentity
 *
 * An API call to find and return a single model instance from the data adapter
 * using the specified criteria.  If an id was specified, just the instance with
 * that unique id will be returned.
 *
 * Optional:
 * @param {String} callback - default jsonp callback param (i.e. the name of the js function returned)
 * @param {*} * - other params will be used as `values` in the create
 */
module.exports = function createRecord (req, res) {

  var Model = actionUtil.parseModel(req);

  // Create data object (monolithic combination of all parameters)
  // Omit the blacklisted params (like JSONP callback param, etc.)
  var data = actionUtil.parseValues(req);


  // Create new instance of model using data from params
  Model.create(data).exec(function created (err, newInstance) {

    // Differentiate between waterline-originated validation errors
    // and serious underlying issues. Respond with badRequest if a
    // validation error is encountered, w/ validation info.
    if (err) return res.negotiate(err);

    // If we have the pubsub hook, use the model class's publish method
    // to notify all subscribers about the created item
    if (req._sails.hooks.pubsub) {
      if (req.isSocket) {
        Model.subscribe(req, newInstance);
        Model.introduce(newInstance);
      }
      Model.publishCreate(newInstance, !req.options.mirror && req);
    }


    // Custom query to populate created record
    var Q = Model.findOne(newInstance.id);
    Q = actionUtil.populateEach(Q, req);
    Q.exec(function foundAgain (err, populatedRecord) {
      if (err) return res.serverError(err);
      if (!populatedRecord) return res.serverError('Could not find record after creating!');
      res.created(populatedRecord);
    });
    
  });
};`

**Comment by mikermcneil** [comment]
@leobreakneck thanks!

**Comment by leobreakneck** [comment]
You're welcome @mikermcneil !
